src/provider/provider_tracking.c

```diff
@@ -7,43 +7,92 @@
  *
  */
 
-#include "provider_tracking.h"
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <umf/memory_pool.h>
+#include <umf/memory_provider.h>
+#include <umf/memory_provider_ops.h>
+
 #include "base_alloc_global.h"
 #include "critnib.h"
 #include "ipc_cache.h"
 #include "ipc_internal.h"
+#include "provider_tracking.h"
 #include "utils_common.h"
 #include "utils_concurrency.h"
 #include "utils_log.h"
 
-#include <umf/memory_pool.h>
-#include <umf/memory_provider.h>
-#include <umf/memory_provider_ops.h>
-
-#include <assert.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
+// TODO: we need to support an arbitrary amount of layers in the future
+#define MAX_LEVELS_OF_MAPS 8
```

Comment:
More descriptive name is needed. Right now it is not clear levels of which map.
Maybe `MAX_LEVELS_OF_ALLOC_SEGMENT_MAP`

Reply:
Done